Add `action` to events

Shatur opened the pull request:
All actions target an entity with a context. This makes sense, since users can attach an input context to their player and react to actions in a convenient way.
However, sometimes it is useful to know which action entity triggered the event - especially if you have multiple identical actions with different bindings.

alice-i-cecile reviewed (approved):
Yes please! Really good change.

Shatur commented:
@alice-i-cecile fixed the bug. It was necessary to resolve this before implementing the feature, so I included it here. I split the work into independent commits and added explanations there. Let's just merge this PR with rebase :slightly_smiling_face:

Shatur commented:
@alice-i-cecile Ah, it was merged because you approved it previously. Feel free to leave suggestions here, I'll open a follow-up if necessary.
